src/lib/settings.ts

```diff
@@ -0,0 +1,104 @@
+import {
+  documentDirectory,
+  writeAsStringAsync,
+  EncodingType,
+  StorageAccessFramework,
+  readAsStringAsync,
+} from 'expo-file-system'
+import * as Sharing from 'expo-sharing'
+import { Alert, Platform } from 'react-native'
+import Toast from 'react-native-root-toast'
+
+import { Secret } from '../types'
+
+const mimeType = 'application/octet-stream'
+
+const androidExport = async (fileName, fileContent) => {
```

Comment:
this function and the one for ios have duplicated error handling. plus, this is somewhat inconsistent with the way we handle errors in the import function. we should identify a single error handling approach and apply it throughout. I would leave the error handling (and showing error messages to the user) to the caller code, meaning the React component.

if you look at the import function, you will also realize there is redundancy because error handling is done both in the function itself AND in the component which calls the function.

I do realize that there are some edge cases (things which are simpler to handle within the function and show a toast when they happen). But apart from those edge cases, let's make sure that exceptions are raised back to the caller and handle them only there.